src/DotFlyer.Api.Tests/APITests.cs

```diff
@@ -0,0 +1,192 @@
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        SMSData smsData = await _cslQueryProvider!
+            .WaitSingleQueryResult<SMSData>($"[\"{SMSTable.Instance.TableName}\"] | where Body == \"{smsMessage.Body}\"", TimeSpan.FromMinutes(10), _cancellationToken);
+
+        smsData.Should().NotBeNull();
+        smsData.Body.Should().Be(smsMessage.Body);
+        smsData.To.Should().Be(smsMessage.To);
+        smsData.TwilioStatusCodeInt.Should().Be(201);
+        smsData.TwilioStatusCodeString.Should().Be("Created");
+        smsData.Tags.Should().Be(JsonSerializer.Serialize(smsMessage.Tags));
+    }
+
+    [TestMethod]
+    public async Task Post_SMS_ShouldReturn_200_When_SenderRoleAndPayloadIsValid()
+    {
+        SMSMessage smsMessage = new()
+        {
+            To = _smsReceiverNumber,
+            Body = Guid.NewGuid().ToString(),
+            Tags = new Dictionary<string, string>()
+            {
+                { "TestName", "DotFlyer API Integration Test" }
+            }
+        };
+
+        using var httpClient = GetHttpClient(await GetSenderAccessTokenAsync());
+
+        var response = await httpClient.PostAsync("dotflyer/sms", GetStringContent(smsMessage), _cancellationToken);
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        SMSData smsData = await _cslQueryProvider!
+            .WaitSingleQueryResult<SMSData>($"[\"{SMSTable.Instance.TableName}\"] | where Body == \"{smsMessage.Body}\"", TimeSpan.FromMinutes(10), _cancellationToken);
+
+        smsData.Should().NotBeNull();
+        smsData.Body.Should().Be(smsMessage.Body);
+        smsData.To.Should().Be(smsMessage.To);
+        smsData.TwilioStatusCodeInt.Should().Be(201);
+        smsData.TwilioStatusCodeString.Should().Be("Created");
+        smsData.Tags.Should().Be(JsonSerializer.Serialize(smsMessage.Tags));
+    }
+
+    [TestMethod]
+    public async Task Post_SMS_ShouldReturn_400_When_SMSSenderRoleAndPayloadIsNotValid()
+    {
+        var smsMessage = new { };
+
+        using var httpClient = GetHttpClient(await GetSMSSenderAccessTokenAsync());
+
+        var response = await httpClient.PostAsync("dotflyer/sms", GetStringContent(smsMessage), _cancellationToken);
+
+        var responseContent = await response.Content.ReadAsStringAsync();
+
+        response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        responseContent.Should().Contain("'To' field is required");
+        responseContent.Should().Contain("'Body' field is required");
+    }
+
+    public static HttpClient GetHttpClient(string? token = null)
+    {
+        HttpClient httpClient = new() { BaseAddress = new($"https://{_apiHost}") };
+
+        if (token != null) httpClient.DefaultRequestHeaders.Authorization = new("Bearer", token);
+
+        return httpClient;
+    }
+
+    public static StringContent GetStringContent(object content) => new(JsonSerializer.Serialize(content), Encoding.UTF8, "application/json");
```

Comment:
Maybe make use of `MediaTypeNames.Application.Json`?
[MediaTypeNames.Application.Json Field](https://learn.microsoft.com/en-us/dotnet/api/system.net.mime.mediatypenames.application.json?view=net-8.0)